### DISCUSSION TITLE
Limit switch question

### DISCUSSION THREAD

**Discussion by rvalotta** [terjeio/grblHAL#317] [Q&A]
So i'm having a little but of a issue with my Z homing switch.  I've always believed that setting it up as a NC was the right way to do it. If a wire breaks or switch dies you get an alarm.   The Z on my cnc refuses to home. The moment is starts going up it thinks the z switch has triggered... I"ve temperarily set it up as NO and it has been working a charm.   Can anyone help me figure out what is going on... That being said the Z limit swtich is the longest wire by far. This is  a 1500x1500 cnc so its over 3m long 

Thanks.

**Comment by terjeio** [comment]
What is the real time report saying about the switch status when triggered/not triggered? Only when triggered should it have a `Z` in the `Pn:` element. If opposite then either `$5` is set to invert the input or the switch is NO (or there is a bug in the code).

**Reply by langwadt** [reply]
I suspect it it noise, I've seen similar on my machine (running my own grbl port) and it seems the wire loop of an NC switch in the bundle together with stepper wires picks up more noise than "open" wires to an NO switch

**Reply by rvalotta** [reply]
I know it is in shielded cable. I haven't grounded the shielding yet I may give that a try as well...
